### PR TITLE
k3s: 1.21.0+k3s1 -> 1.21.2+k3s1

### PR DESCRIPTION
**Pull request by superherointj**
k3s: 1.21.0+k3s1 -> 1.21.2+k3s1

@euank I had to update the patch for building properly.
Please, review my changes. Thanks.

### CODE REVIEW

File: pkgs/applications/networking/cluster/k3s/patches/0002-Add-nixpkgs-patches.patch [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
-These patches let us re-use the upstream build scripts when building for
-nix.
----
- scripts/build       | 20 ++++----------------
- scripts/package-cli | 10 ++++++----
- 2 files changed, 10 insertions(+), 20 deletions(-)
-
+-Subject: [PATCH 2/2] Add nixpkgs patches
+-Original patch by: Euan Kemp <euank@euank.com>
+-Adapted by: superherointj
```
Comment (mohe2015) [inline pkgs/applications/networking/cluster/k3s/patches/0002-Add-nixpkgs-patches.patch:3]: Probably doesn't matter but I would keep the original attributes and just add a `Co-authored-by: superherointj`.

Comment (superherointj) [reply pkgs/applications/networking/cluster/k3s/patches/0002-Add-nixpkgs-patches.patch:3]: The original information does not match how current patch was generated (commit revision and dates won't match). Information on lines changed is unavailable. I have manually added this header.

**Review by mohe2015 — APPROVED** [review]
LGTM! Updated my laptop k3s and test-deployed a helm chart. Didn't update my pi though as I was too lazy to recompile on it / use cross-compilation.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130025` at 5494ab9f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>k3s</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

util-linux is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/networking/cluster/k3s/default.nix:259:3:
```
    |
259 |   buildInputs = [
    |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `git`.
Near pkgs/applications/networking/cluster/k3s/default.nix:10:3:
```
   |
10 | , git
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/networking/cluster/k3s/patches/0002-Add-nixpkgs-patches.patch (hunk 1 above)
Comment (euank) [reply pkgs/applications/networking/cluster/k3s/patches/0002-Add-nixpkgs-patches.patch:3]: I wouldn't worry much about the headers here; nixpkgs git history has the canonical authorship info, and it just happens that git format-patch with email headers is the easiest way to copy around and apply patch files.

What you've got looks good to me!

**Review by euank — APPROVED** [review]
Thanks for the bump, lgtm

**Pull request merged**